### ISSUE TITLE
memory is keeping increase until crash

### ISSUE THREAD

**Issue by vincentzzh33** [rcrowley/go-metrics#244]
newStandardMeter newEWMA these two memory is keeping increase until crash when i using sarama as httpserver to receive kafka request

**Comment by shriram1993** [comment]
Yes!! We are also facing memory leak in **newStandardMeter** & **newEWMA**. Attached pprof graph screenshot for reference. Please help.
![screenshot-localhost-8081-2019 05 01-12-14-14](https://user-images.githubusercontent.com/17744393/57008030-b2fe6500-6c0a-11e9-96aa-308e97f25ab5.png)

**Comment by shriram1993** [comment]
> Yes!! We are also facing memory leak in **newStandardMeter** & **newEWMA**. Attached pprof graph screenshot for reference. Please help.
> ![screenshot-localhost-8081-2019 05 01-12-14-14](https://user-images.githubusercontent.com/17744393/57008030-b2fe6500-6c0a-11e9-96aa-308e97f25ab5.png)

If you need more info, I can provide.

**Comment by shriram1993** [comment]
Any Update on this?

**Comment by luzhuqun** [comment]
Any Update?

**Comment by michaelschlies** [comment]
Is anyone that is running into this registering their `Meter`s with the registry in go-metrics? I just refactored registry to use a sync.Map instead of a mutex managed map and removing its switch() for what to allow to register and my memory leaks have completely stopped.

@luzhuqun @shriram1993 @vincentzzh33

**Comment by omgommt** [comment]
Identified the issue:
Its happening because we are passing NewMetric object with every metric log. Instead it should be function, which will be called when new Metric object is required.

Before:
.RcRegistry.GetOrRegister(timerName, rcmetric.NewTimer()).(rcmetric.Timer)

After: (Correct)
.RcRegistry.GetOrRegister(timerName, rcmetric.NewTimer).(rcmetric.Timer)

**Comment by omgommt** [comment]
> Any Update on this?

Identified the issue:
Its happening because we are passing NewMetric object with every metric log. Instead it should be function, which will be called when new Metric object is required.

Before:
.RcRegistry.GetOrRegister(timerName, rcmetric.NewTimer()).(rcmetric.Timer)

After: (Correct)
.RcRegistry.GetOrRegister(timerName, rcmetric.NewTimer).(rcmetric.Timer)
